src/js/components/Button/Button.js

```diff
@@ -84,17 +84,30 @@ class Button extends Component {
     } = this.props;
     const { hover } = this.state;
 
-    let buttonIcon = icon;
     // only change color if user did not specify the color themselves...
-    if (primary && icon && !icon.props.color) {
+    let buttonIcon = icon;
+    // nonprimary button icon styling with custom theme
+    if (!primary && icon && !icon.props.color) {
       buttonIcon = cloneElement(icon, {
-        color:
-          theme.global.colors.text[
-            isDarkBackground(this.props) ? 'dark' : 'light'
-          ],
+        color: normalizeColor(theme.button.color, theme),
       });
     }
-
+    // primary button icon styling with custom theme
+    if (primary && icon && !icon.props.color) {
+      if (theme.button.color !== undefined) {
```

Comment:
Why are we using `theme.button.color` and not `theme.button.primary.color` for the case of a primary button?

Reply:
The theme.button.primary.color when I used that it took that color as the background of the button not the color of the icon.

Reply:
![Screen Shot 2019-04-30 at 9 04 36 AM](https://user-images.githubusercontent.com/42451602/56971858-3fcff280-6b27-11e9-8010-9c5c8d988dd6.png)

Reply:
lgtm